kiwi/repository/__init__.py

```diff
@@ -41,9 +41,9 @@ class Repository(object):
         list of custom package manager arguments
         to setup the repository
     """
-    def __new__(self, root_bind, package_manager, custom_args=None):
+    def __new__(self, root_bind, package_manager, custom_args=None, exclude_docs=False):
```

Comment:
for passing additional arguments the class provides the custom_args parameter. I think exclude_docs should be part of custom_args not an additional argument to the constructor. We might end up with a continuously growing param list which I'd like to avoid especially because we try to provide a common interface with Repository and PackageManager for different implementations (zypper, yum, apt... maybe more) and so I just like to pass in a dictionary with customization values and keep the class design as stable as possible.

That way the class implementing the functionality can work over the custom_args and provide information if the data is used, ignored, not implemented by us, or by the tool without changing the interface all the time we add new values